go.mod

```diff
@@ -5,19 +5,21 @@ go 1.13
 require (
 	github.com/apparentlymart/go-dump v0.0.0-20190214190832-042adf3cf4a0 // indirect
 	github.com/aws/aws-sdk-go v1.25.3 // indirect
-	github.com/google/go-cmp v0.3.1 // indirect
-	github.com/gophercloud/gophercloud v0.17.0
+	github.com/gophercloud/gophercloud v0.18.0
 	github.com/gophercloud/utils v0.0.0-20210530213738-7c693d7efe47
 	github.com/hashicorp/go-retryablehttp v0.6.4
 	github.com/hashicorp/go-version v1.2.0
-	github.com/hashicorp/hcl v1.0.0 // indirect
 	github.com/hashicorp/hil v0.0.0-20190212132231-97b3a9cdfa93 // indirect
 	github.com/hashicorp/terraform-plugin-sdk v1.0.0
-	github.com/mattn/go-colorable v0.1.1 // indirect
-	github.com/mattn/go-isatty v0.0.7 // indirect
+	github.com/metal3-io/baremetal-operator v0.0.0-20211201170610-92ffa60c683d
+	github.com/metal3-io/baremetal-operator/apis v0.0.0
+	github.com/metal3-io/baremetal-operator/pkg/hardwareutils v0.0.0
 	github.com/ulikunitz/xz v0.5.6 // indirect
 	github.com/vmihailenco/msgpack v4.0.4+incompatible // indirect
 	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
-	golang.org/x/net v0.0.0-20191126235420-ef20fe5d7933 // indirect
-	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
+)
+
+replace (
+	github.com/metal3-io/baremetal-operator/apis => github.com/metal3-io/baremetal-operator/apis v0.0.0-20211201170610-92ffa60c683d
```

Comment:
We might skip replace unless we use downstream(openshift/baremetal-operator)?

Reply:
If I don't use replace. It will fail to import apis.
```
$ go mod tidy
go: github.com/metal3-io/baremetal-operator@v0.0.0-20211201170610-92ffa60c683d requires
        github.com/metal3-io/baremetal-operator/apis@v0.0.0: reading github.com/metal3-io/baremetal-operator/apis/apis/go.mod at revision apis/v0.0.0: unknown revision apis/v0.0.0
```

Reply:
I mean, can't we import this directly `github.com/metal3-io/baremetal-operator/apis v0.0.0-20211201170610-92ffa60c683d` ?

Reply:
As we talked in Slack, the reason of this problem is first `github.com/metal3-io/baremetal-operator v0.0.0-20211201170610-92ffa60c683d` needs to be removed from go.mod